core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1459,12 +1462,56 @@ private void unparseRegexMatchCount(SqlWriter writer, SqlCall call,
     writer.endFunCall(arrayLengthFrame);
   }
 
-  private void unparseRegexpExtractAll(SqlWriter writer, SqlCall call,
+  private void unparseRegexpExtract(SqlWriter writer, SqlCall call,
+      int leftPrec, int rightPrec) {
+    int indexOfRegexOperand = 1;
+    SqlWriter.Frame regexpExtractAllFrame = writer.startFunCall("REGEXP_EXTRACT");
+    List<SqlNode> operandList = call.getOperandList();
+    unparseRegexFunctionsOperands(writer, leftPrec, rightPrec, indexOfRegexOperand, operandList);
+    writer.endFunCall(regexpExtractAllFrame);
+  }
+
+  private void unparseRegexpReplace(SqlWriter writer, SqlCall call,
+      int leftPrec, int rightPrec) {
+    int indexOfRegexOperand = 1;
+    SqlWriter.Frame regexpReplaceFrame = writer.startFunCall("REGEXP_REPLACE");
+    List<SqlNode> operandList = call.getOperandList();
+    unparseRegexFunctionsOperands(writer, leftPrec, rightPrec, indexOfRegexOperand, operandList);
+    writer.endFunCall(regexpReplaceFrame);
+  }
+
+  private void unparseRegexpInstr(SqlWriter writer, SqlCall call,
```

Comment:
possibly the redundant code could have been made a single function with additional argument.

Reply:
As discussed, we'll address this in another MR.